packages/core/ui-settings/core-ui-settings-common/src/ui_settings.ts

```diff
@@ -92,6 +92,10 @@ export interface UiSettingsParams<T = unknown> {
    * scoped to a namespace. The default value is 'namespace'
    */
   scope?: UiSettingsScope;
+  /**
+   * Specifies whether the setting should be displayed in the Advanced settings app.
+   */
+  allowlisted?: boolean;
```

Comment:
just a nit, we try to keep core unaware of other parts of kibana like advanced settings. I might consider renaming this to be something more generic like `readonly`

Reply:
I forgot about the readonly param and looked into it further. It appears to hide the setting from the UI only, but doesn't seem to actually prevent changing the setting via the `/internal/kibana/settings` API.

Thinking aloud:

Maybe it would make more sense to leverage that existing flag, and add enforcement to it in the API (e.g. if a setting is `readonly: true`, you cannot set it). Then the API in core could basically allow manipulating this value, something like `coreSetup.uiSettings.setReadOnly(['*'])` and then `coreSetup.uiSettings.setWritable([...])`? Need to think about that some more, @clintandrewhall any opinions?

We'd need to consider whether it is safe to roll out enforcement to self-managed since it would affect any of the existing settings that are readonly, and customers may be relying on being able to change those programmatically via API. Using a separate flag like `frozen` or `enabled` would solve this problem, it's just sort of redundant I guess since `readonly` is the behavior we care about here.

There might not be a good way around having another flag, but long-term I'd suggest we push to only use `readonly` for this purpose.

Reply:
> Then the API in core could basically allow manipulating this value, something like `coreSetup.uiSettings.setReadOnly(['*'])` and then `coreSetup.uiSettings.setWritable([...])`?

@luke I think this makes it a bit confusing about what's been called where, and the status of the any/all of the settings.

Another option would be to allow an API on the service to define the _nature_ of readonly.  For example, something like `coreSetup.uiSettings.setReadOnlyState('hidden' | 'frozen')`.  Then the behavior in fully-managed doesn't change.  Setting an allow-list could toggle this, but at least we don't change the nature of `readonly`.

Reply:
> I think this makes it a bit confusing about what's been called where, and the status of the any/all of the settings.

Yeah that's fair, I'm struggling to think of what the simplest API would be here. As we discussed offline, perhaps we stick with `setAllowlist` and as a side effect enforce it at the API level. Then in 9.0, we can change the behavior of `readonly` in classic so that we are always enforcing `readonly` in the APIs as well. This is where I'd like to get more 👀 from the core team.